### PR TITLE
Added JSON scenario

### PR DESCRIPTION
**Pull request by davidfowl**
- Added support for --webHost and --kestrelThreadCount
- Added JSON benchmarks to repo.targets
- Added Sockets and Libuv based transports to repo.targets
- Added JSON to BenchmarkApplication using Utf8Json (do we need to do something here before committing this?) @Eilon 

/cc @DamianEdwards @benaadams

### CODE REVIEW

File: src/PlatformBenchmarks/BenchmarkApplication.cs [hunk 1]
```diff
@@ -90,17 +117,39 @@ private static void PlainText(PipeWriter pipeWriter)
             writer.Write(DateHeader.HeaderBytes);
 
             // Content-Type header
-            writer.Write(_headerContentTypeText);
+            writer.Write(_headerContentTypeJson);
```
Comment (DamianEdwards) [inline src/PlatformBenchmarks/BenchmarkApplication.cs]: Would stackalloc'ing the object here be of any benefit?

Comment (davidfowl) [reply src/PlatformBenchmarks/BenchmarkApplication.cs]: Which object?

Comment (DamianEdwards) [reply src/PlatformBenchmarks/BenchmarkApplication.cs]: Bleh, wrong line. The obejct that gets serialized to JSON.

**Comment by Eilon** [comment]
@davidfowl yes, and I completed the OSS registration for `Utf8Json`. Thanks!

**Comment by davidfowl** [comment]
> @davidfowl yes, and I completed the OSS registration for Utf8Json. Thanks!

Do I wait for confirmation?

File: build/repo.targets [hunk 2]
```diff
@@ -2,9 +2,10 @@
 
   <PropertyGroup>
     <PlaintextJobs>-j "$(RepositoryRoot)src/Benchmarks/benchmarks.plaintext.json"</PlaintextJobs>
-    <PlaintextPlatformJobs>-j "$(RepositoryRoot)src/PlatformBenchmarks/benchmarks.json"</PlaintextPlatformJobs>
+    <PlaintextPlatformJobs>-j "$(RepositoryRoot)src/PlatformBenchmarks/benchmarks.plaintext.json"</PlaintextPlatformJobs>
     <HtmlJobs>-j "$(RepositoryRoot)src/Benchmarks/benchmarks.html.json"</HtmlJobs>
     <JsonJobs>-j "$(RepositoryRoot)src/Benchmarks/benchmarks.json.json"</JsonJobs>
+    <PlatformJsonJobs>-j "$(RepositoryRoot)src/PlatformBenchmarks/benchmarks.json.json"</PlatformJsonJobs>
```
Comment (sebastienros) [inline build/repo.targets]: Inconsistent naming [Plaintext]Platform -> [Json]Platform

File: build/repo.targets [hunk 3]
```diff
@@ -83,11 +87,12 @@
 
   <ItemGroup>
 
-    <PhysicalScenarios Include="-n PlaintextPlatform --webHost KestrelLibuv $(PlaintextPlatformJobs)" />
+    <PhysicalScenarios Include="-n PlaintextPlatform --webHost KestrelSockets $(PlaintextPlatformJobs)" />
     <PhysicalScenarios Include="-n Plaintext --webHost KestrelSockets $(PlaintextJobs)" />
     <PhysicalScenarios Include="-n Plaintext --webHost KestrelSockets $(PlaintextJobs)" Description="Baseline" AspNetCoreVersion="Current" RuntimeVersion="Current"/>
     
     <!-- Optimal ThreadCount for current hardware -->
+    <PhysicalScenarios Include="-n PlaintextPlatform --webHost KestrelLibuv --kestrelThreadCount 2 $(PlaintextPlatformJobs)" />
```
Comment (sebastienros) [inline build/repo.targets]: We'll need to note this change (thread count) as we are already storing results for a week on an automatic count

**Comment by davidfowl** [comment]
@sebastienros 🆙 📅

**Comment by Eilon** [comment]
@davidfowl :

```
I completed the OSS registration
  ^^^^^^^^^
```

**Pull request merged**